### PR TITLE
Fix eslint on save

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Fix eslint on save.

### QA steps

- Open vscode. Make a change that the linter complains about.
- Save the file.
- The eslint error should get fixed (if it knows how to).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-  "cSpell.words": ["lifecycles", "unmount"]
+  "cSpell.words": ["lifecycles", "unmount"],
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
```
Comment (huwshimi) [inline .vscode/settings.json:4]: Should we enable this do you think?

Comment (Caleb-Ellis) [reply .vscode/settings.json:4]: I'm personally not a fan. Formatting on save always catches me by surprise - sometimes I just want to save a file so I can close it, then I can come back later to fix it up and have it be exactly as I left it, warts and all. Plus I've baked it into my muscle memory to format before saving "for real" anyway.

Though I'm not gonna be around to be very slightly annoyed by it so don't listen to me at all hahaha

Comment (huwshimi) [reply .vscode/settings.json:4]: I guess I was thinking about https://github.com/canonical-web-and-design/maas-ui/pull/3922 and the import order linting which is much easier for the computer to do. Right now I've got a shortcut mapped to fix eslint errors and I can happily keep running that.

Comment (petermakowski) [reply .vscode/settings.json:4]: I always had this enabled in my user settings anyway. Let's get this merged!

**Review by petermakowski — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (huwshimi) [reply .vscode/settings.json:4]: > I always had this enabled in my user settings anyway. Let's get this merged!

I looked, but I couldn't find a way to do that! What do I need to do? That seems a bit better than having to turn this on for everyone.

Comment (petermakowski) [reply .vscode/settings.json:4]: Adding this option to global vscode user preferences (`Preferences: Open Settings (JSON)`  via the command palette) seemed to do the trick for me.

Comment (huwshimi) [reply .vscode/settings.json:4]: Thanks! I didn't know about that.

**Pull request merged**